src/useTheme.ts

```diff
@@ -0,0 +1,76 @@
+import { useEffect, useState } from 'react';
+import type { ComponentTheme, UseThemeReact } from './types';
+import { useOnchainKit } from './useOnchainKit';
+
+export function useTheme(): UseThemeReact {
```

Comment:
Theme selection is very complex since we have so many variations based on Theme and Mode while also keeping a clean input for the developers - one `default` theme.